node/src/blockchain/blockchain_interface.rs

```diff
@@ -1395,120 +1788,153 @@ mod tests {
     }
 
     #[test]
-    fn send_transaction_fails_on_badly_prepared_consuming_wallet_without_secret() {
+    fn signing_error_ends_iteration_over_accounts_after_detecting_first_error_which_is_then_propagated_all_way_up_and_out(
+    ) {
+        let transport = TestTransport::default();
+        let batch_payable_tools = BatchPayableToolsMock::<TestTransport>::default()
+            .sign_transaction_result(Err(Web3Error::Signing(
+                secp256k1secrets::Error::IncorrectSignature,
+            )))
+            //we return after meeting the first result
+            .sign_transaction_result(Err(Web3Error::Internal));
+        let mut subject = BlockchainInterfaceNonClandestine::new(
+            transport,
+            make_fake_event_loop_handle(),
+            Chain::PolyMumbai,
+        );
+        subject.batch_payable_tools = Box::new(batch_payable_tools);
+        let recipient = Recorder::new().start().recipient();
+        let consuming_wallet = make_wallet("consume, you greedy fool!");
+        let nonce = U256::from(123);
+        let accounts = vec![make_payable_account(5555), make_payable_account(6666)];
+
+        let result = subject.send_payables_within_batch(
+            &consuming_wallet,
+            111,
+            nonce,
+            &recipient,
+            &accounts,
+        );
+
+        assert_eq!(
+            result,
+            Err(PayablePaymentError::UnusableWallet(
```

Comment:
Can you make it clearer, either in comments or preferably through code, why this error is the result of injecting the error above with `sign_transaction_result()`?